File: client/src/boundary/interactor/GameD2BConverter.ts
import { Game } from "domain/index";
import { BoundaryGame } from "boundary/model/index";
import { Converter } from "utils/Converter";
import { BoardD2BConverter } from "./BoardD2BConverter";

export class GameD2BConverter extends Converter<Game, BoundaryGame> {
  private boardConverter: BoardD2BConverter;

  constructor(boardConverter: BoardD2BConverter) {
    super();
    this.boardConverter = boardConverter;
  }

  convert(game: Game): BoundaryGame {
    return new BoundaryGame(
      game.id,
      game.active,
      this.boardConverter.convert(game.board)
    );
  }
}

File: server/battleships/use_cases/implementation/src/main/java/eu/optas/use_cases/implementation/GameD2BConverter.java
package eu.optas.use_cases.implementation;

import eu.optas.domain.Game;
import eu.optas.use_cases.api.BoundaryGame;
import eu.optas.utils.Converter;

public class GameD2BConverter extends Converter<Game, BoundaryGame> {
    private BoardD2BConverter boardD2BConverter;

    public GameD2BConverter(BoardD2BConverter boardD2BConverter) {
        this.boardD2BConverter = boardD2BConverter;
    }

    @Override
    public BoundaryGame convert(Game game) {
        return new BoundaryGame(
                game.getId(),
                game.getState(),
                boardD2BConverter.convert(game.getBoard()),
                game.getHitsRemaining(),
                game.getShipsDestroyed()
        );
    }
}
